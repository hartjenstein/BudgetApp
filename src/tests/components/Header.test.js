import React from 'react';
import { shallow } from 'enzyme';
//import ReactShallowRenderer from 'react-test-renderer/shallow'
import Header from '../../components/Header';


// react-test-renderer
//shallow rendering test in contrast to full-DOM_rendering only the component,
//not its child components
// we cant test events like buttons with react renderer thats why we are going to use enzyme instead
test('should render Header correctly', () => {
    const wrapper = shallow(<Header />);   
    expect(wrapper).toMatchSnapshot();

    /* const renderer = new ReactShallowRenderer();
    renderer.render(<Header/ >);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
    console.log(renderer.getRenderOutput()); */
    //expect(wrapper.find('h1').length).toBe(1);
});