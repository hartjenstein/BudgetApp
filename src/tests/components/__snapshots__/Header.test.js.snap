// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header
  className="header"
>
  <div
    className="content-container"
  >
    <div
      className="header__content"
    >
      <Link
        className="header__title"
        replace={false}
        to="/dashboard"
      >
        <h1>
          Budget Control
        </h1>
      </Link>
      <div
        className="header__link-container"
      >
        <button
          className="button button--link"
          onClick={[Function]}
        >
          Logout
        </button>
        <Link
          className="header__link align-bottom"
          replace={false}
          to="/dsgvo"
        >
          <p>
            Datenschutzerkl√§rung
          </p>
        </Link>
      </div>
    </div>
  </div>
</header>
`;
